include/string_utils.h

```diff
@@ -375,4 +379,24 @@ std::optional<float> parse_prefixed_percentage(const char prefix, const std::str
 // returns value only if succeeded
 std::optional<int> to_int(const std::string& value);
 
+template <typename... Args>
+std::string string_format(const std::string& format, Args... args) noexcept
```

Comment:
Yeah, I'm just not that familiar with them. I'll give it a go.

Reply:
Ok, `string_view` does _not_ have the terminating zero nor a `c_str()` method and `snprint()` needs a zero-terminated C string. So just using `std::string` for the format strings, otherwise we'd need to copy the contents of `string_view` anyway. Lesson learned: `string_view` and C string functions don't really mix.